### PR TITLE
feat(CI): Automatically assign milestones to pull-requests

### PR DESCRIPTION
**Pull request by srebhan**
This adds a Github action to assign milestones to pull-requests **after** merging them. The assignment is based on the label of the PR in the following way:

- PRs with one of the labels `fix`, `chore`, `docs` and `dependencies` are assigned to the next **bugfix** release `x.y.z+1`.
- PRs having the label `breaking` are assigned to the next **major** release `x+1.0.0`.
- All remaining PRs are assigned to the next **minor** release `x.y+1.0`.

The labels are evaluated in the above order and are assigned to the first match. However, to avoid ambiguities make sure that we do not have colliding labels!

**Please make sure the labels for a PR are correct **before** merging!**

### CODE REVIEW

**Comment by powersj** [comment]
> PRs with one of the labels fix, chore, docs and dependencies are assigned to the next bugfix release x.y.z+1.

This is not always the case. For example if you merge a fix at say v1.26.4, there isn't a v1.26.5. It would go into v1.27.0

> PRs having the label breaking are assigned to the next major release x+1.y.z.

There are no planned future major releases, so I don't find this helpful. This would only cause questions about when 2.x will come out. When in reality it would probably go out with the next minor release.

**Comment by Hipska** [comment]
Do I need to approve the enabling of this task or actually check the task source code?

**Comment by srebhan** [comment]
> This is not always the case. For example if you merge a fix at say v1.26.4, there isn't a v1.26.5. It would go into v1.27.0

@powersj right and I took care of this. If a milestone exists, say we are are v1.26.3 and v1.26.4 **exists and is open**, the PR will be assigned to v1.26.4. However, if we are at v1.26.4 as in your example, we will not create a v1.26.5 milestone in the project. In this case, the Github action will fall back to the **fallback** setting which is `minor` in our case and thus the PR gets assigned to v1.27.0.

> There are no planned future major releases, so I don't find this helpful. This would only cause questions about when 2.x will come out. When in reality it would probably go out with the next minor release.

Right, then we should not set any of the labels _and_/_or_ listed. As we do not create a `v2.0.0` milestone, we also never assign anything to v2.0.0, the PRs will go to the next minor milestone. If this is unwanted, we should remove the milestones from the few PRs afterwards...

**Comment by srebhan** [comment]
> Do I need to approve the enabling of this task or actually check the task source code?

Both, if possible. ;-) Otherwise, please state what you approved. ;-)

File: .github/workflows/milestones.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        uses: srebhan/label-milestone-action@v1.0.0
+        id: assign-milestone
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          bugfix-labels: 'fix,chore,docs,dependencies'
+          minor-labels: 'feat'
+          major-labels: 'breaking change'
+          fallback: 'minor'
+      - name: Show milestone
+        run: echo "Assigned milestone is ${{ steps.assign-milestone.outputs.milestone }}"
```
Comment (Hipska) [inline .github/workflows/milestones.yml:24]: End the file with a newline maybe?

**Review by Hipska — APPROVED** [review]
I approve addition of this action, looks handy. Did not check action source code.

But does this action create the milestones as well or do they need to exist first?

**Comment by srebhan** [comment]
> But does this action create the milestones as well or do they need to exist first?

The milestones have to exist. So we should create the next milestone during the release process.

**Review by Hipska — APPROVED** [review]

**Comment by Hipska** [comment]
Great, this will improve visibility a lot!

And what about PRs like this one? Should those also be linked to bugfix version milestone?

**Comment by srebhan** [comment]
Those are difficult to handle. I could introduce a "ignore-labels" option to leave out stuff that is infrastructure only. @powersj what do you think?

**Comment by powersj** [comment]
> @powersj what do you think?

If we are going to start to track when PRs land then I think having infrastructure PRs tracked as part of milestones is important as well. So no need to ignore them.

**Review by powersj — APPROVED** [review]

**Pull request merged**